src/Control/Eff/Internal.hs

```diff
@@ -109,19 +109,17 @@ comp (Arrs f) (Arrs g) = Arrs (f >< g)
 (^|>) :: Arrs r a b -> Arr r b c -> Arrs r a c
 (Arrs f) ^|> g = Arrs (f |> g)
 
--- | The Eff monad (not a transformer!). It is a fairly standard coroutine monad
--- where the type @r@ is the type of effects that can be handled, and the
--- missing type @a@ (from the type application) is the type of value that is
--- returned.  It is NOT a Free monad! There are no Functor constraints.
+-- | The monad that all effectful computation in this library bases on.
 --
--- The two constructors denote the status of a coroutine (client): done with the
--- value of type a, or sending a request of type Union r with the continuation
--- Arrs r b a. Expressed another way: an `Eff` can either be a value (i.e.,
--- 'Val' case), or an effect of type @`Union` r@ producing another `Eff` (i.e.,
--- 'E' case). The result is that an `Eff` can produce an arbitrarily long chain
--- of @`Union` r@ effects, terminated with a pure value.
+-- The type `Eff r a` has 2 argumens where `r` is a type-level list of effects
+-- that can occur inside an effectful computation and `a` is the computation's
+-- result.
+--
+-- This monad is used via the  functions provided by  concrete effect-types
+-- implemented in this or another library.
```

Comment:
@power-fungus which "another" library? would it be better perhaps to reference some of the elements you mention in the README here? specifically, the notions of
- operations denoting the use of (or "requesting") the effect
- and handlers which denote the interpretation of (or "responding" to) the effect

then you can be precise in what you mean here